lib/websocket.js

```diff
@@ -940,7 +940,9 @@ function socketOnEnd() {
   // the closing handshake but unlike `WebSocket.CLOSING` it does not prevent
   // `WebSocket.prototype.close()` from sending a close frame.
   //
-  websocket._readyState = -2;
+  if (websocket._readyState === WebSocket.OPEN) {
```

Comment:
Why not removing the `receiverOnFinish()` listener from `receiverOnError()` instead?